File: javascript/grains/grains.js
var BigInt = require('./big-integer');
var Grains = function(){}

Grains.prototype.square = function(squareNumber) {
  var total = BigInt(0);
  if (typeof squareNumber != 'object') {
    squareNumber = BigInt(squareNumber);
  }

  if (squareNumber == 1) {
    total = total.add(1);
  } else {
    var partial = BigInt(this.square(squareNumber.minus(1)));
    total = total.add(partial.multiply(2));
  }

  return total.toString();
}

Grains.prototype.total = function() {
  var total = BigInt(0);
  for (var i = 64; i > 0; i--) {
    var square = BigInt(this.square(i));
    total = total.add(square);
  }

  return total.toString();
}

module.exports = Grains

File: c/grains/grains.c
#include "grains.h"

uint64_t square(uint8_t index)
{
    uint64_t value_square = 1;
    if (index == 0 || index > 64){
        return 0;
    }
    for (int i = 2 ; i <= index ; i++){
        value_square = value_square * 2;
    }

    return value_square;
}

uint64_t total(void)
{
    uint64_t total = 0;
    for (int i = 1 ; i <= 64 ; i++){
        total += square(i);
    }
    return total;
}